requirements/constraints.txt

```diff
@@ -43,6 +43,9 @@ edx-search<2.0.0
 # We expect v2.0.0 to introduce large breaking changes in the feature toggle API
 edx-toggles<2.0.0
 
+# We expect v3.3.0 to introduce breaking changes in the testing API
```

Comment:
1. This should be <4.0.0. For semantic versions, backward incompatible means major version bump.
2. Based on what I saw in monitoring, it seems like the legacy classes are used outside of tests too, but not sure.